source/dofs/dof_handler_policy.cc

```diff
@@ -3004,10 +3004,10 @@ namespace internal
 
 
 
-        template <int dim, int spacedim>
+        template <int dim, int spacedim, typename DoFHandlerType>
```

Comment:
Maybe also require `DoFHandlerType` to have `spacedim`and `dim` parameters, i.e. `template <int, int> class DoFHandlerType`?